samples/sample_common/src/vaapi_device.cpp

```diff
@@ -353,7 +353,7 @@ mfxStatus CVAAPIDeviceWayland::Init(mfxHDL hWindow, mfxU16 nViews, mfxU32 nAdapt
 
 mfxStatus CVAAPIDeviceWayland::RenderFrame(mfxFrameSurface1 * pSurface, mfxFrameAllocator * /*pmfxAlloc*/)
 {
-    uint32_t drm_format = 0;
+    uint32_t drm_format = pSurface->Info.FourCC;
```

Comment:
here ```pSurface``` might be ```NULL```. Please, move check placed at line 361 to the beginning of the function